src/Core/Framework/DataAbstractionLayer/DataAbstractionLayerException.php

```diff
@@ -772,4 +773,16 @@ public static function foreignKeyNotFoundInDefinition(string $association, strin
             ['association' => $association, 'entityDefinition' => $entityDefinition]
         );
     }
+
+    /**
+     * @internal
+     */
+    public static function deprecatedDefinitionCall(): self
```

Comment:
we should already mark it as deprecated, as the whole exception can be removed with 6.7 right?